### PR TITLE
DBSCAN broadcast fix

### PR DESCRIPTION
**Pull request by Er1cCheng**
Fix DBSCAN bug where data is not broadcasted to each GPU worker and stored separately in different GPU memory

### CODE REVIEW

**Comment by Er1cCheng** [comment]
build

File: python/src/spark_rapids_ml/clustering.py [hunk 1]
```diff
@@ -959,6 +960,13 @@ def _cuml_fit(
             dbscan.n_cols = params[param_alias.num_cols]
             dbscan.dtype = np.dtype(dtype)
 
+            # get_logger(self.__class__).info("Row", concated.shape)
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/clustering.py:963]: best to delete this commented code.

Comment (Er1cCheng) [reply python/src/spark_rapids_ml/clustering.py:963]: Oops, forgot to delete the code used to detect the out of memory stuff. Sorry about that.

**Comment by Er1cCheng** [comment]
build

**Comment by Er1cCheng** [comment]
build

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**